.github/workflows/pr_check_workflow.yml

```diff
@@ -1,25 +1,107 @@
 # This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
 
-name: Node.js CI
+name: Build and test
 
 on:
   push:
     branches: [ main ]
   pull_request:
     branches: [ main ]
 
-jobs:
-  build:
+env:
+  CACHE_NAME: osd-node-modules
+  TEST_BROWSER_HEADLESS: 1
+  CI: 1
+  GCS_UPLOAD_PREFIX: fake
+  TEST_OPENSEARCH_DASHBOARDS_HOST: localhost
+  TEST_OPENSEARCH_DASHBOARDS_PORT: 6610
+  TEST_OPENSEARCH_TRANSPORT_PORT: 9403
+  TEST_OPENSEARCH_PORT: 9400
 
+jobs:
+  build-lint-test:
     runs-on: ubuntu-latest
+    name: Build and Verify
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "10.24.1"
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Setup Yarn
+        run: |
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.10
 
+      - name: Run bootstrap
+        run: yarn osd bootstrap
+
+      - name: Run linter
+        run: yarn lint
+
+      - name: Run unit tests
+        run: node scripts/jest --ci --colors --maxWorkers=10
+        env:
+          SKIP_BAD_APPLES: true
```

Comment:
Does this mean we will skip some unit tests below for now? Any plan to bring them back?

Reply:
only for github actions. locally if this env var is not set it will still run them. the next step would be to 'quarantine' the bad apples and fix them or drop them. more likely to fix them since they work well on good hardware but work bad on bad hardware.